### PR TITLE
cpu: forward SIGINT/SIGTERM to remote commands

### PR DESCRIPTION
**Pull request by Lencerf**
Signed-off-by: Changyuan Lyu <changyuanl@google.com>

### CODE REVIEW

File: cmds/cpu/cpu.go [hunk 1]
```diff
@@ -149,17 +159,50 @@ func newCPU(host string, args ...string) error {
...
+	defer close(errChan)
+
+	go func() {
+		verbose("start")
+		if err := c.Start(); err != nil {
+			errChan <- fmt.Errorf("Start: %v", err)
+			return
+		}
+		verbose("wait")
+		if err := c.Wait(); err != nil {
```
Comment (rminnich) [inline cmds/cpu/cpu.go]: another way to do this:
errChan <- c.Wait()
has the same effect.

File: cmds/cpu/cpu.go [hunk 2]
```diff
@@ -149,17 +159,50 @@ func newCPU(host string, args ...string) error {
...
+			var sigErr error
+			switch sig {
+			case syscall.SIGINT:
+				sigErr = c.Signal(ossh.SIGINT)
+			case syscall.SIGTERM:
+				sigErr = c.Signal(ossh.SIGTERM)
+			}
+			verbose("signal %v sent to %q", sig, c.Args[0])
+			if sigErr != nil {
+				err = fmt.Errorf("sending %v to %q: %v", sig, c.Args[0], sigErr)
```
Comment (rminnich) [inline cmds/cpu/cpu.go]: I would argue that we should not quit here, log an error and continue?

File: server/server.go [hunk 3]
```diff
@@ -62,6 +62,23 @@ func handler(s ssh.Session) {
...
 	cmd := command(a[0], a[1:]...)
+
+	sigChan := make(chan ssh.Signal, 1)
+	defer close(sigChan)
+	s.Signals(sigChan)
+	go func() {
+		for signal := range sigChan {
+			switch signal {
+			case ssh.SIGTERM:
+				cmd.Process.Signal(syscall.SIGTERM)
```
Comment (rminnich) [inline server/server.go]: should we check for an error here?

Comment (Lencerf) [reply server/server.go]: Done

File: cmds/cpu/cpu.go (hunk 2 above)
Comment (Lencerf) [reply cmds/cpu/cpu.go]: Done.

File: cmds/cpu/cpu.go (hunk 1 above)
Comment (Lencerf) [reply cmds/cpu/cpu.go]: Done.

**Review by rminnich — APPROVED** [review]

**Pull request merged**